src/rprblender/engine/viewport_engine_2.py

```diff
@@ -43,12 +43,13 @@ def __init__(self, rpr_engine):
         self.resolve_lock = threading.Lock()
 
     def stop_render(self):
-        self.rpr_context.set_render_update_callback(None)
-
+        self.restart_render_event.set()
+        
         super().stop_render()
 
         self.resolve_event.set()
         self.resolve_thread.join()
+        self.rpr_context.set_render_update_callback(None)
```

Comment:
It raises error here: `AttributeError: 'NoneType' object has no attribute 'set_render_update_callback'` because super().stop_render() sets self.rpr_context = None.

Reply:
This function can be written in this way
```
    def stop_render(self):
        self.is_finished = True
        self.restart_render_event.set()
        self.resolve_event.set()
        self.sync_render_thread.join()
        self.resolve_thread.join()

        self.rpr_context.set_render_update_callback(None)
        self.rpr_context = None
        self.image_filter = None

```